soroban-sdk/src/storage.rs

```diff
@@ -45,7 +54,16 @@ use crate::{
 /// # fn main() { }
 /// ```
 #[derive(Clone)]
-pub struct Storage(Env);
+pub struct Storage {
+    env: Env,
+    mode: StorageMode,
+}
+
+#[derive(Clone)]
+enum StorageMode {
+    Persistent,
```

Comment:
High level question - do you think the separation of persistent vs temp storage at this level can be confusing for users? I'm wondering if users will run into scenarios where an entry is in one group, but the other is checked by mistake.  Temp storage is a advanced feature so this shouldn't be an issue but curious what you think.

Reply:
The public interface is `env.storage()` or `env.temp_storage()`, so they're pretty distinct. Also mismatched lookups would be just a bug - not sure there is anything special about it vs any other bug.